app/models/game.rb

```diff
@@ -34,6 +34,40 @@ def populate_board!
   end
 
   def check?
-    # code
+    white_king = King.find_by(color: 'white')
+    black_king = King.find_by(color: 'black')
+
+    pawns = Pawn.all
+    knights = Knight.all
+    bishops = Bishop.all
+    rooks = Rook.all
+    queens = Queen.all
+
+    pawns.each do |pawn|
+      return true if pawn.valid_move?(white_king.position_x, white_king.position_y) && pawn.color != white_king.color
+      return true if pawn.valid_move?(black_king.position_x, black_king.position_y) && pawn.color != black_king.color
+    end
+
+    knights.each do |knight|
+      return true if knight.valid_move?(white_king.position_x, white_king.position_y) && knight.color != white_king.color
+      return true if knight.valid_move?(black_king.position_x, black_king.position_y) && knight.color != black_king.color
+    end
+
+    bishops.each do |bishop|
+      return true if bishop.valid_move?(white_king.position_x, white_king.position_y) && bishop.color != white_king.color
+      return true if bishop.valid_move?(black_king.position_x, black_king.position_y) && bishop.color != black_king.color
+    end
+
+    rooks.each do |rook|
+      return true if rook.valid_move?(white_king.position_x, white_king.position_y) && rook.color != white_king.color
+      return true if rook.valid_move?(black_king.position_x, black_king.position_y) && rook.color != black_king.color
+    end
+
+    queens.each do |queen|
+      return true if queen.valid_move?(white_king.position_x, white_king.position_y) && queen.color != white_king.color
+      return true if queen.valid_move?(black_king.position_x, black_king.position_y) && queen.color != black_king.color
+    end
```

Comment:
Notice how each of these blocks are the same.

Could dry this up by doing.

```Ruby
pieces.where.not(type: 'King').each do |piece|
  ##...
end  
```

I'm not sure if this is valid Rails 5 code: `pieces.where.not(type: 'King')`

Give it a try in a console 😄

Reply:
`pieces.where.not(type: 'King')` did seem to work. I updated this part of the `check?` method. Please let us know if this is right.

Reply:
Rubocop changed `each` to `find_each` in `pieces.where.not(type: 'King').find_each do |piece|`.

Reply:
Changing this to `find_each` broke the `check?` method. I changed this back to `each` by doing the following, and it seemed to work:
`non_king_pieces = pieces.where.not(type: 'King')`
`non_king_pieces.each do |piece|`